frontend/src/store/chats/reducer.ts

```diff
@@ -64,10 +64,17 @@ const reducer = createReducer(initialState, (builder) => {
     state.currentChatMessagesDataStatus = DataStatus.REJECTED;
   });
 
-  builder.addCase(createMessage.fulfilled, (state, action) => {
+  builder.addCase(createMessage.fulfilled, (state, { payload }) => {
     state.dataStatus = DataStatus.FULFILLED;
-    state.currentChatMessages = [...state.currentChatMessages, action.payload];
-    state.currentChatId = action.payload.chatId;
+    state.currentChatMessages = [...state.currentChatMessages, payload];
+    state.currentChatId = payload.chatId;
+    state.lastMessages = state.lastMessages.map((lastMessage) => {
+      if (lastMessage.chatId === payload.chatId) {
+        return payload;
+      }
+
+      return lastMessage;
+    });
   });
```

Comment:
I would prefer to use findIndex, because it's hard to understand what's going on

Reply:
I prefer `.map` 😼